oeps/oep-0026-arch-xapi-integration.rst

```diff
@@ -0,0 +1,483 @@
+for many different purposes, it can be used to train adaptive engines as shown in the
+diagram below.
+
+.. image:: oep-0026/adaptive_learning_lrs_basic.png
+   :alt: The diagram above is enhanced with a new LRS component that receives events
+    from the Open edX "Eventing" component and is accessed by adaptive engines for
+    training purposes, using xAPI for both transactions.
+
+In the short term, however, we will not implement our own LRS since the initial integration
+efforts do not require this (as shown in the first diagram above).
+
+Statement API
+=============
+
+xAPI specifies 4 standard REST-ful JSON payload APIs, of which only the `Statement API`_
+is needed for our purposes as that API is sufficient for tracking learning activity.
+
+The Statement API has essentially the following parts: `xAPI Actor`_, `xAPI Verb`_,
+`xAPI Object`_, `xAPI Context`_, and `xAPI Result`_.
+
+.. _Statement API: https://xapi.com/statements-101/
+
+xAPI Actor
+~~~~~~~~~~
+Although the **Actor** field can be either an *Agent* or a *Group*, we will primarily
+support only the *Agent* type, which is used for individuals performing an activity
+(`xAPI Verb`_ on an `xAPI Object`_).
+
+An Actor can be identified using `Friend of a Friend (FOAF)`_ vocabulary with either:
+(1) `email address`_, (2) `hash of email address`_, (3) `OpenID URI`_, or (4) `account`_
+with a *homepage*-scoped identifier.  One of these is sent along with the Actor's "name".
+To be mindful of learner privacy, we will initially take a conservative approach
+and only send #3, with an Open edX anonmyzed unique identifier of the learner
+(`Anonymized User ID`_).
+
+In the future, if certain external systems require `Personally Identifiable Information
+(PII)`_, like the learner's email address or name, then those may be conditionally sent
+with appropriate permissions. Adaptive engines, however, do not need PII.
+
+Initially, we will exclude the "name" field. However, if we find that xAPI JSON parsers
+assume this field always exists, then we can include the field but provide a non-PII
+value, such as a copy of the `Anonymized User ID`_.
+
+Anonymized User ID
+^^^^^^^^^^^^^^^^^^
+An *Anonymized User ID* will be generated as a SHA-256 hash of the Open edX user_id, 
+which is the **id** value of the user's row in the `Django auth_user`_ table in the
+Open edX LMS.
+
+As mentioned in `Using consistent user identifiers in Segment events`_, the other user
+identifiers in the Open edX platform aren't perfectly suited for inclusion in externally
+sent events:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Identifier
+     - Limitations
+   * - username
+     - Often contains PII, such as user's name or year of birth.
+   * - LMS user_id
+     - Information leakage of auth_user table with easily guessable values; tied down
+       to implementation of the housing database.
+   * - email address
+     - Is PII and modifiable by the user.
+   * - anonmyous user_id
```

Comment:
`s/anonmyous/anonymous/`

Reply:
fixed. thanks.  (I'm looking for a spell checker now for VSCode.  I have one for PyCharm, but not one for VSCode, which I've now switched to using primarily.)